pkg/ebpf/c/tracee.bpf.c

```diff
@@ -3450,22 +3455,42 @@ int BPF_KPROBE(trace_do_exit)
     return events_perf_submit(&data, DO_EXIT, code);
 }
 
-/* uprobe_syscall_trigger submit to the buff the syscalls function handlers
- * address from the syscall table. the syscalls are stored in map which is
- * syscalls_to_check_map and the syscall-table address is stored in the
- * kernel_symbols map.
- */
+// uprobe_syscall_trigger submit to the buff the syscalls function handlers
+// address from the syscall table. the syscalls are stored in map which is
+// syscalls_to_check_map and the syscall-table address is stored in the
+// kernel_symbols map.
+
 SEC("uprobe/trigger_syscall_event")
 int uprobe_syscall_trigger(struct pt_regs *ctx)
 {
-    u64 caller_ctx_id;
-#if defined(bpf_target_x86)
-    caller_ctx_id = ctx->bx;
-#elif defined(bpf_target_arm64)
-    bpf_probe_read(&caller_ctx_id, 8, ((void *) ctx->sp) + 16);
-#else
-    return 0;
-#endif
+    u64 magic_num = 0;
+    u64 caller_ctx_id = 0;
+
+    // clang-format off
```

Comment:
@AsafEitani could you, please, check this PR for the hooked events (with diamorphine and any other tests you got) and let me know if it works for you ? A good way of testing both arm64 situations is to have go set to 1.17 and then go set to 1.18 (you can link /usr/bin/go to /usr/lib/go-1.17/bin/go in debian/ubuntu, for example).

Thank you!